src/anemoi/models/interface/__init__.py

```diff
@@ -97,17 +127,54 @@ def predict_step(self, batch: torch.Tensor) -> torch.Tensor:
         torch.Tensor
             Predicted data.
         """
-        batch = self.pre_processors(batch, in_place=False)
 
         with torch.no_grad():
 
             assert (
                 len(batch.shape) == 4
             ), f"The input tensor has an incorrect shape: expected a 4-dimensional tensor, got {batch.shape}!"
+            x = self.pre_processors_state(batch[:, 0 : self.multi_step, ...], in_place=False)
+
             # Dimensions are
-            # batch, timesteps, horizonal space, variables
-            x = batch[:, 0 : self.multi_step, None, ...]  # add dummy ensemble dimension as 3rd index
+            # batch, timesteps, horizontal space, variables
+            x = x[..., None, :, :]  # add dummy ensemble dimension as 3rd index
+            if self.prediction_strategy == "tendency":
+                tendency_hat = self(x)
+                y_hat = self.add_tendency_to_state(x[:, -1, ...], tendency_hat)
+            else:
+                y_hat = self(x)
+                y_hat = self.post_processors_state(y_hat, in_place=False)
```

Comment:
These multiple `if self.prediction_strategy == "tendency"`  are adding complexity in the main (and currently unique)  AnemoiModelInterface class. This will create technical debt and make the code less flexible and more difficult to maintain.

If you want an alternative, this is what you could do:

Since the current  AnemoiModelInterface does not have the behaviour you want, you can instead to create another class that behave as you like (copy-paste the whole  AnemoiModelInterface in to  MyTendencyTrainingAnemoiModelInterface) and use this one instead.

Just after copy-pasting, you realise that you have duplicated code between AnemoiModelInterface and MyTendencyTrainingAnemoiModelInterface. To avoid this, you can put the common code into a mother class BaseAnemoiModelInterface.

Reply:
I see that @JesperDramsch had the same opinion on this https://github.com/ecmwf/anemoi-models/pull/23#issuecomment-2352336126

Reply:
Sounds good! I’ve already worked on additional interfaces for a different purpose on a local branch, so this should be straightforward to implement. However, if we split the interface, we’ll likely need to divide the current "AnemoiLightningModule" in anemoi-training into two separate modules: one for ForecastingState (ForecastingStateLightningModule) and another for ForecastingTendency (ForecastingTendencyLightningModule).

At the moment, the existing AnemoiLightningModule (which is tied to the current interface) contains logic for handling both state and tendency steps forward.